subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/AvailableToolChains.java

```diff
@@ -336,6 +340,23 @@ public String getId() {
         }
 
         public abstract String getUnitTestPlatform();
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(displayName);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            final InstalledToolChain other = (InstalledToolChain) obj;
+            return Objects.equal(this.displayName, other.displayName);
+        }
```

Comment:
Is this to ensure that we discover only a single tool chain with the same display name? If so, I think we should call out that behavior explicitly, say, but instead putting the discovered toolchains into a `Map` indexed by the display name.

Reply:
That is not for finding the same toolchain with the same display name but for finding the same installation in different positions in the path. On my machine for example I found the same Gcc installation twice. I changed the logic to only add Gcc installations for different files, and removed the equals/hashCode implementation from `ToolChainCandidate`.